### DISCUSSION TITLE
Switching locale while inside iterations is not working

### DISCUSSION THREAD

**Discussion by IkboljonMe** [amannn/next-intl#1508] [Q&A]
### Description

Long in short, <Link> or useRouter is not working to switch locale inside iteration. As far as I know it should work, I think it is bug so here in my code iterations:
```
  {languages.map((language) => (
      <div>
            <Link
            key={language.code}
            href="/" // Base URL; it will localize based on the selected language
            locale="uz" // Set the locale to the selected language
            className={classNames("flex items-start gap-3 text-white text-4xl cursor-pointer hover:scale-105 transition-transform", {
              "bg-gray-800": language.code === selectedLanguage
            })}
            onClick={() => onSelectLanguage(language.code)} // Update selected language in state
          >
            <span>{language.flag}</span>
            <span>{language.label}</span>
          </Link>
      </div>
        ))}
```
I tried to switch with help of `useRouter`,

### Verifications

- [X] I've verified that the problem I'm experiencing isn't covered in [the docs](https://next-intl-docs.vercel.app/).
- [X] I've searched for similar, existing issues on [GitHub](https://github.com/amannn/next-intl/issues) and [Stack Overflow](https://stackoverflow.com/search?q=next-intl).
- [X] I've compared my app to [a working example](https://next-intl-docs.vercel.app/examples) to look for differences.

### Mandatory reproduction URL

App Router

### Reproduction description

Steps to reproduce:
1. Open reproduction
2. Click on …
3. See error: …
Done

### Expected behaviour

It should work as expected

**Comment by amannn** [comment]
Moving to a discussion since it's a usage question. Also, consider consulting the examples.

**Reply by doksara** [reply]
I have the same issue. If it's an "usage question", would you care to elaborate what seems to be the root cause then? @amannn
